src/nomad_simulations/physical_property.py

```diff
@@ -0,0 +1,231 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from typing import Any, Optional
+
+from nomad import utils
+from nomad.datamodel.data import ArchiveSection
+from nomad.metainfo import (
+    Quantity,
+    SubSection,
+    SectionProxy,
+    Reference,
+    Section,
+    Context,
+    MEnum,
+)
+from nomad.metainfo.metainfo import DirectQuantity, Dimension, _placeholder_quantity
+from nomad.datamodel.metainfo.basesections import Entity
+
+from .variables import Variables
+from .numerical_settings import SelfConsistency
+
+
+# We add `logger` for the `PhysicalProperty.variables_shape` method
+logger = utils.get_logger(__name__)
+
+
+class PhysicalProperty(ArchiveSection):
```

Comment:
As of now, `PhysicalProperty` has many Area C specific fields that don't need to be a part of the underlying base class. I would say that if we want to add such a base class to the core packages, it should only contain `value`, `rank` and `variables` in the first version. This will simplify the discussions a lot, but also does not stop you from extending the base class to fit your needs.

Reply:
> has many Area C specific fields

I very much disagree with this. Would you mind elaborating? Maybe the only one being specific is the `self_consistency_ref` and `is_scf_converged` quantities. Anything else is pretty general.